File: result/translate/transcoder/python2java/java/CHECK_POSSIBLE_TRANSFORM_ONE_STRING_ANOTHER.java
public static boolean f_gold ( String s1 , String s2 ) {
  int n = s1 . length ( ) ;
  int m = s2 . length ( ) ;
  boolean [ ] [ ] dp = ( boolean [ ] [ ] ) new boolean [ m + 1 ] [ n + 1 ] ;
  dp [ 0 ] [ 0 ] = true ;
  for ( int i = 0 ;
  i < s1 . length ( ) ;
  i ++ ) {
    for ( int j = 0 ;
    j < s2 . length ( ) + 1 ;
    j ++ ) {
      if ( ( dp [ i ] [ j ] ) != null ) {
        if ( ( ( j < s2 . length ( ) ) && ( s1 . charAt ( i ) . toUpperCase ( ) . equals ( s2 . charAt ( j ) ) ) ) ) {
          dp [ i + 1 ] [ j + 1 ] = true ;
        }
        if ( ( s1 . charAt ( i ) . isUpperCase ( ) == false ) && ( s1 . charAt ( i + 1 ) . toUpperCase ( ) . equals ( s2 . charAt ( j ) ) ) ) {
          dp [ i + 1 ] [ j ] = true ;
        }
      }
    }
  }
  return ( dp [ n ] [ m ] ) ;
}

File: result/translate/transcoder/java2cpp/c++/CHECK_POSSIBLE_TRANSFORM_ONE_STRING_ANOTHER.cpp
bool f_gold ( string s1 , string s2 ) {
  int n = s1 . size ( ) ;
  int m = s2 . size ( ) ;
  bool dp [ n + 1 ] [ m + 1 ] ;
  for ( int i = 0 ;
  i <= n ;
  i ++ ) {
    for ( int j = 0 ;
    j <= m ;
    j ++ ) dp [ i ] [ j ] = false ;
  }
  dp [ 0 ] [ 0 ] = true ;
  for ( int i = 0 ;
  i < s1 . size ( ) ;
  i ++ ) {
    for ( int j = 0 ;
    j <= s2 . size ( ) ;
    j ++ ) {
      if ( dp [ i ] [ j ] ) {
        if ( j < s2 . size ( ) && ( toupper ( s1 [ i ] ) == s2 [ j ] ) ) {
          dp [ i + 1 ] [ j + 1 ] = true ;
        }
        if ( ! toupper ( s1 [ i ] ) ) {
          dp [ i + 1 ] [ j ] = true ;
        }
      }
    }
  }
  return ( dp [ n ] [ m ] ) ;
}